[WIP] Improve Endscreen

xeruf opened the pull request:
fixes #5, fixes #22

Commit:
Move Endscreen from Viewer to Game

Fixes #22

Commit:
Refactor Game

Commit:
Unify styling of start-button and endscreen and put the controls on top

Commit:
Fix going back to Turn 0

Previously, the back button did not go from Turn 1 to 0, because 0 was
interpreted as false and thus 1 was used instead.

Commit:
Eliminate repeated status updates in Game

Commit:
Remove debug messages from Game

Xerus2000 commented:
Ich habe die commits semantisch gruppiert, sodass es auch sinnvoll sein könnte sie einzeln zu reviewen.

SKoschnicke commented:
Wenn ein menschliche Spieler Teilnimmt, springt die Auswahl des Zielfeldes nach kurzer Zeit wieder zurueck auf die Auswahl des zu bewegenden Fisches. Das macht es sehr schwierig, Zuege einzugeben. Passiert das bei dir auch, @Xerus2000 ?

Xerus2000 commented:
Ja. Ich schaue jetzt endlich mal danach :sweat_smile:

Commit:
Fix repeated Viewer updates

The Viewer was updated every time nextTurn was called, because turnTotal
was always updated. Now it is updated properly.

Xerus2000 commented:
@SKoschnicke sollte jetzt funktionieren ;)
